lib/backend/sqlite.cc

```diff
@@ -194,8 +209,19 @@ static int sqlite_fini(rpmdb rdb)
 	    rdb->db_opens--;
 	} else {
 	    if (sqlite3_db_readonly(sdb, NULL) == 0) {
+
 		sqlexec(sdb, "PRAGMA optimize");
 		sqlexec(sdb, "PRAGMA wal_checkpoint = TRUNCATE");
+
+		int max_size = rpmExpandNumeric("%{?_sqlite_vacuum}");
+		if (max_size <= 0)
+		    max_size = 20*1024*1024;
```

Comment:
I initially misread this, which is why I didn't comment earlier: %_sqlite_vacuum should not be in *bytes*, that makes no sense at all. The kind of ballpark range we care about is megabytes, but if for testing purposes it needs to be kilos then so be it.